news/templates/news/user_news.html

```diff
@@ -1,46 +1,38 @@
 {% load i18n %}
 {% load news %}
 
-{% for item in alerts %}
-<div class="alert alert-{{ item.alert }}">
-  <h4>
-    {{ item.title|safe }}
-    {% if not item|is_published:now %}<small><span class="label label-default">{% trans "Appears" %} {{ item.publish }}</span></small>{% endif %}
-    {% if is_course_staff %}<small><span class="label label-default">{{ item.audience|news_audience }}</span></small>{% endif %}
-  </h4>
-  <div>
-    {{ item.body|safe }}
-  </div>
-</div>
-{% endfor %}
 
 {% if news %}
 <div class="panel panel-default news-panel">
   <div class="panel-heading">
-    <h4 class="panel-title">{% trans "Course news" %}</h4>
+    <h3 class="panel-title">{% trans "Course news" %}</h3>
   </div>
   <div class="list-group">
     {% for item in news %}
-    <a class="list-group-item{% if item.collapsed %} collapsed{% endif %}" href="#news{{ item.id }}" data-toggle="collapse" role="button" aria-controls="news{{ item.id }}" aria-expanded="{% if item.collapsed %}false{% else %}true{% endif %}">
+    <div class="{% if item.pin %}pinned-{% endif %}list-group-item">
       <h4 class="list-group-item-heading">
-        <span class="caret"></span>
-        {{ item.title|safe }} <small>{{ item.publish }}</small>
-        {% if not item|is_published:now %}<small><span class="label label-default">{% trans "Appears later" %}</span></small>{% endif %}
-        {% if is_course_staff %}<small><span class="label label-default">{{ item.audience|news_audience }}</span></small>{% endif %}
+        {{ item.title|safe }}
+        {% if not item|is_published:now %}
+          <small class="future-instance">{% trans "Published on" %}: {{ item.publish }}</small>
+        {% else %}
+          <small class="current-instance">{{ item.publish }}</small>
+        {% endif %}
+        {% if is_course_staff %}<small><span class="label label-primary">{{ item.audience|news_audience }}</span></small>{% endif %}
+        {% if is_course_staff and item.pin %}<small><span class="glyphicon glyphicon-pushpin"></span></small>{% endif %}
       </h4>
-    </a>
-    <div class="list-group-item-text collapse{% if not item.collapsed %} in{% endif %}" id="news{{ item.id }}">
-      {{ item.body|safe }}
+      <div class="list-group-item-text">
+        {{ item.body|safe }}
+      </div>
     </div>
     {% if item.begin_more %}
-    <a class="list-group-item" href="#more-news" data-toggle="collapse" role="button" aria-controls="more-news" aria-expanded="false">
-      {% trans "Show more" %}
+    <a class="folding-list-group-item collapsed" href="#more-news" data-toggle="collapse" role="button" aria-controls="more-news" aria-expanded="false">
```

Comment:
I think this solution is fine for now. If this would be reworked, then having two lists "visible" and "hidden" and two for loops for the collapsed area nd for the visible. Both for loops would use a partial for the news div. Sadly, this pattern is inefficient with django templates. Would be fine with Jinja. Hence, let's keep this complicated solution.

Reply:
I tried to simplify this a bit. The  this version allowed me to remove also the for-loop from news.py and still works for all the different course index -options:
https://github.com/apluslms/a-plus/pull/570/commits/03b9cecbba023f49826f94f3460bcaab1e449c2c